components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -961,6 +961,59 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             ))).filter(e => e !== undefined) as WhitelistedRepository[];
     }
 
+    public async getSuggestedContextURLs(ctx: TraceContext): Promise<string[]> {
+        const user = this.checkUser("getSuggestedContextURLs");
+        const suggestions: string[] = [];
+
+        // Fetch all data sources in parallel for maximum speed (don't await before `Promise.allSettled(promises)` below!)
+        const promises = [];
+
+        // Example repositories
+        promises.push(this.getFeaturedRepositories(ctx).then(exampleRepos => {
+            // log('got example repos', exampleRepos);
+            exampleRepos.forEach(r => suggestions.push(r.url));
+        }));
+
+        // User repositories
+        user.identities.forEach(identity => {
+            const provider = {
+                'Public-GitLab': 'gitlab.com',
+                'Public-GitHub': 'github.com',
+                'Public-Bitbucket': 'bitbucket.org',
+            }[identity.authProviderId];
+            if (!provider) {
+                return;
+            }
+            promises.push(this.getProviderRepositoriesForUser(ctx, { provider }).then(userRepos => {
```

Comment:
I don't think reusing this method is a good idea as it limits the list of repos to the repos that have the app installed. Also fetches projects etc. IMHO this should be entirely about proposing great matches for a user, no matter an app is installed. We will grow out of what this method does pretty soon anyway.

Reply:
Good point, thanks. Added a `TODO` comment as a first start.